### DISCUSSION TITLE
Students can't accept assignment

### DISCUSSION THREAD

**Discussion by victorialindsaymcgee** [community/community#122130] [GitHub Education]
### Hi everyone,

I've tried using Classroom for the first time, and my students are getting this message when trying to accept assignments:

**This assignment can no longer be accepted. Please contact your teacher. They might need to edit or recreate the assignment.**

This happens for me too when I try to go in as a student. The assignment dates are still well in the future (end of June). Any suggestions?

**Comment by charlestanev** [comment]
Hi Victoria, thanks for raising this issue. It sounds frustrating, especially when you're ready to use Classroom and this happens. I've had a similar problem with GitHub Education before, and what helped was double-checking a few things:

Assignment Settings: Did you set the correct due dates and ensure the assignment is published? Sometimes minor details like these can cause unexpected issues.
Repository Permissions: If you have a template repository, make sure it hasn't been deleted or made private.
Account Permissions: Have you verified that all students have the necessary GitHub permissions to accept assignments? Sometimes, students need a specific type of account or a GitHub Education discount to access certain features.
If these don't help, reaching out to GitHub Education Support might be your best bet. They can provide detailed guidance and potentially fix any backend issues causing this problem.

I'm curious if others in the community have encountered this problem and what solutions worked for them.

**Comment by vnijs** [comment]
I'm getting the same issue. Did you find a fix?

**Comment by acorbin3** [comment]
I have created a few assignments yesterday with no issues. I just created 2 assignments today and have an issue. 

I have a feeling I was going too fast to accept the assignment after creating it to cause the issue, but that's only an assumption.

**Reply by acorbin3** [reply]
I just recreated 2 assignments and they still have an issue

https://classroom.github.com/classrooms/92205571-spc-adam-corbin-cop2801-fall-2024/assignments/assignment-5v2
https://classroom.github.com/a/4Ww6gQmx

https://classroom.github.com/classrooms/92205571-spc-adam-corbin-cop2801-fall-2024/assignments/assignment-6v2
https://classroom.github.com/a/0MVikHTY

**Comment by delcin-raj** [comment]
Yes facing the same issue.
Wondering if it has to do with recent github failure

**Reply by LouisTsai-Csie** [reply]
@delcin-raj I have the same issue for almost 8 hours. What happened to GitHub today? Did the server crash?

**Reply by OlivierJM** [reply]
Yeah it was down at some point yesterday and today, It could be related to this.

**Comment by rafaelcorsi** [comment]
many students impacted by this this week

**Comment by RobinGTM** [comment]
Same issue here, people can't accept assigments, and I can't edit the assignments (I have access to the editing page, but I get errors when clicking "Update assignment" to validate my edits.

Error when trying to accept: same as @victorialindsay.
Errors when trying to edit:
- "Starter code repository must have at least one commit. Select a repository with commits or create the assignment without starter code." when I edit and immediately click "Update assignment" without touching anything
- "Starter code repo could not be copied to your organization" when I change the starter code to the original repo with which I created the assignment (which is a template repo).

I don't think this is my mistake, since I did the same operations without errors yesterday.

[edit] typo

**Reply by RobinGTM** [reply]
Problem seems fixed now! Without doing anything. Maybe it was a backend issue.

**Comment by bryanosborne** [comment]
Seeing all the errors posted by other users:
- This assignment can no longer be accepted. Please contact your teacher. They might need to edit or recreate the assignment. 
- Starter code repository must have at least one commit. Select a repository with commits or create the assignment without starter code.

Also have the undocumented feature of not being able to delete the assignment to start again - receiving the following:

406 Request not acceptable
The requested resource is unavailable in the format requested.

Not helpful with the Fall semester starting soon...please help!

**Comment by MrZ85** [comment]
Same here - unable to create assignments for my class of 400 students starting in less than two weeks. Help!

**Reply by MrZ85** [reply]
My problem seems to have resolved on its own. Hope it stays that way.

**Comment by hernanconiglio** [comment]
Hi! My students are not being able to accept assignments! Help^2 !!!
Error: "This assignment can no longer be accepted. Please contact your teacher. They might need to edit or recreate the assignment."

**Comment by tomastrivino** [comment]
Same here – hundreds of students affected

**Comment by hernanconiglio** [comment]
The solution I found is to make a commit (a trivial change in the README) on the repository that is generated from the assignment (not the public tamplate repository) and the problem was solved.

**Reply by jessicadesilva** [reply]
That worked for me, thank you!

**Reply by bryanosborne** [reply]
Excellent workaround! Thanks!

**Reply by acorbin3** [reply]
Just checked and it worked for me too!

**Reply by jyounggo** [reply]
works for me as well. Thanks!

**Reply by GSibinger** [reply]
That worked. Thanks a lot!

**Comment by jessicadesilva** [comment]
Same here. I tried to delete the assignment and recreate it, but attempting to delete the assignment gives a 406 error.

<img width="772" alt="Screenshot 2024-08-15 at 12 22 46 PM" src="https://github.com/user-attachments/assets/5054b228-1bd1-44c2-8dea-75eb8c6783d7">

**Comment by LouisTsai-Csie** [comment]
Can anyone help me with this one, thank you very much

https://github.com/orgs/community/discussions/135880#discussioncomment-10347664

**Comment by hashiprobr** [comment]
Same issue here, reported by students yesterday. Will try the suggestion by @hernanconiglio and report back.

**Comment by CodyHenrichsen-CTEC** [comment]
The trivial edit did not work. @samus-aran  Have you heard about this??? You were very helpful last year! My morning class was able to access earlier today but my entire afternoon session gets the failed repo error.

**Reply by tomastrivino** [reply]
This is absolutely ridiculous coming from such a big company as GitHub.

**Reply by chrissphinx** [reply]
@tomastrivino honestly it's par for the course when a company has an "educational" division or offering. I can say the experience is identical with Texas Instruments.

**Comment by wonsunahn** [comment]
I am experiencing this exact same issue today.  Students cannot accept the assignment and I cannot delete the assignment either due to the 406 error.  It used to work well until a couple of days ago.  Anybody else with the same issue?
